scripts/version_calc.py

```
@@ -0,0 +1,312 @@
+        warnings=warnings,
+    )
+
+
+# ---------------------------------------------------------------------------
+# Git I/O (thin shell, not unit-tested)
+# ---------------------------------------------------------------------------
+
+def git_latest_semver_tag() -> str:
+    """Find the latest semver tag reachable from HEAD, excluding prereleases."""
+    result = subprocess.run(
+        ["git", "tag", "--list", "v*", "--sort=-v:refname"],
```

Ruff (0.15.9)

[error] Starting a process with a partial executable path

(S607)